python/triton/runtime/jit.py

```diff
@@ -301,7 +301,7 @@ def __init__(self, fn, version=None, do_not_specialize=None):
         self.has_defaults = any(v.default != inspect._empty for v in signature.parameters.values())
         # specialization hints
         self.do_not_specialize = [] if do_not_specialize is None else do_not_specialize
-        self.do_not_specialize = set([self.arg_names.index(arg) if isinstance(arg, str) else arg for arg in self.do_not_specialize])
+        self.do_not_specialize = {self.arg_names.index(arg) if isinstance(arg, str) else arg for arg in self.do_not_specialize}
```

Comment:
Although there is nothing syntactically wrong with this code, it is hard to read and can be simplified to a set comprehension. Using set comprehension is more performant since there is no need to create a transient list.